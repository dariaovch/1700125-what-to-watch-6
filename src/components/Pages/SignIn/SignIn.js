import React from 'react';
import controllersImage from 'src/images/controllersImage.svg';
import Header from 'src/components/Layout/Header/Header';
import Footer from 'src/components/Layout/Footer/Footer';

function SignIn() {
  return (
    <>
      <div className="visually-hidden">
        <img src={controllersImage} />
      </div>

      <div className="user-page">
        <Header theme="user" hasAvatar={false} />

        <div className="sign-in user-page__content">
          <form action="#" className="sign-in__form">
            <div className="sign-in__fields">
              <div className="sign-in__field">
                <input className="sign-in__input" type="email" placeholder="Email address" name="user-email" id="user-email" />
                <label className="sign-in__label visually-hidden" htmlFor="user-email">Email address</label>
              </div>
              <div className="sign-in__field">
                <input className="sign-in__input" type="password" placeholder="Password" name="user-password" id="user-password" />
                <label className="sign-in__label visually-hidden" htmlFor="user-password">Password</label>
              </div>
            </div>
            <div className="sign-in__submit">
              <button className="sign-in__btn" type="submit">Sign in</button>
            </div>
          </form>
        </div>

        <Footer />
      </div>
    </>
  );
}

export default SignIn;
