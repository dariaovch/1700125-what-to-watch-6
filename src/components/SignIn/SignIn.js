import React from 'react';
import Controllers from 'src/images/controllers.svg';
import Header from 'src/components/Header/Header.js';
import Footer from 'src/components/Footer/Footer.js';

function SignIn() {
  return (
    <>
      <div className="visually-hidden">
        {/* <!-- inject:svg --> */}
        <img src={Controllers} />
        {/* <!-- endinject --> */}
      </div>

      <div className="user-page">
        <Header theme="user" hasAvatar={false} />

        <div className="sign-in user-page__content">
          <form action="#" className="sign-in__form">
            <div className="sign-in__fields">
              <div className="sign-in__field">
                <input className="sign-in__input" type="email" placeholder="Email address" name="user-email" id="user-email" />
                <label className="sign-in__label visually-hidden" htmlFor="user-email">Email address</label>
              </div>
              <div className="sign-in__field">
                <input className="sign-in__input" type="password" placeholder="Password" name="user-password" id="user-password" />
                <label className="sign-in__label visually-hidden" htmlFor="user-password">Password</label>
              </div>
            </div>
            <div className="sign-in__submit">
              <button className="sign-in__btn" type="submit">Sign in</button>
            </div>
          </form>
        </div>

        <Footer />
      </div>
    </>
  );
}

export default SignIn;
