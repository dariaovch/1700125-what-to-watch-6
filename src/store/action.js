export const ActionType = {
  FILTER_LIST: `filterMovies/filterListByGenre`,
  LOAD_MOVIES: `data/loadMovies`,
  REDIRECT_TO_ROUTE: `app/redirectToRoute`,
  REQUIRE_AUTH: `user/requireAuth`,
  GET_USER_DATA: `user/getUserData`,
  GET_MOVIE_DATA: `data/getMovieData`,
};

export const ActionCreator = {
  filterListByGenre: (item, movies) => ({
    type: ActionType.FILTER_LIST,
    payload: {
      genre: item,
      movies: item.id === `00` ? movies : movies.filter((movie) => movie.genre === item.title),
    },
  }),
  loadMovies: (movies) => ({
    type: ActionType.LOAD_MOVIES,
    payload: movies,
  }),
  requireAuth: (status) => ({
    type: ActionType.REQUIRE_AUTH,
    payload: status,
  }),
  getUserData: (userData) => ({
    type: ActionType.GET_USER_DATA,
    payload: userData,
  }),
  getMovieData: (movieData) => ({
    type: ActionType.GET_MOVIE_DATA,
    payload: movieData,
  }),
  redirectToRoute: (url) => ({
    type: ActionType.REDIRECT_TO_ROUTE,
    payload: url,
  }),
};

