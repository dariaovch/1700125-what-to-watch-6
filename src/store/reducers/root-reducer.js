import {combineReducers} from 'redux';
import {data} from 'src/store/reducers/data/data';
import {user} from 'src/store/reducers/user/user';

export const NameSpace = {
  DATA: `DATA`,
  USER: `USER`,
};

export default combineReducers({
  [NameSpace.DATA]: data,
  [NameSpace.USER]: user,
});
