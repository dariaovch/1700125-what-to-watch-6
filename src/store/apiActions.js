import {ActionCreator} from "src/store/action";

export const fetchMovies = () => (dispatch, _getState, api) => (
  api.get(`/films`)
    .then(({data}) => dispatch(ActionCreator.loadMovies(data)))
);
