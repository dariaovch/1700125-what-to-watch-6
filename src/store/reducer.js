import {ActionType} from 'src/store/action';

const initialState = {
  genre: `All Genres`,
  movies: [],
  isDataLoaded: false,
};

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case ActionType.LOAD_MOVIES:
      return {
        ...state,
        movies: action.payload,
        isDataLoaded: true,
      };
    case ActionType.FILTER_LIST:
      return {
        ...state,
        genre: action.payload.genre,
        movies: action.payload.movies,
      };

  }

  return state;
};

export {reducer};

