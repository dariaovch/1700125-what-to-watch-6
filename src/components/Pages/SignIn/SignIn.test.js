import React from 'react';
import {Router} from 'react-router-dom';
import {render} from '@testing-library/react';
import {createMemoryHistory} from 'history';
import configureStore from 'redux-mock-store';
import SignIn from 'src/components/Pages/SignIn/SignIn';
import {Provider} from 'react-redux';
import {NameSpace} from 'src/store/reducers/rootReducer';
import {AuthStatus} from 'src/store/auth';

test(`SignIn component should render correctly`, () => {
  const history = createMemoryHistory();
  const mockStore = configureStore({});
  const store = mockStore({
    [NameSpace.USER]: {
      authStatus: AuthStatus.NO_AUTH,
    },
  });

  const {container} = render(
      <Provider store={store}>
        <Router history={history}>
          <SignIn />
        </Router>
      </Provider>);

  expect(container).toMatchSnapshot();
});
