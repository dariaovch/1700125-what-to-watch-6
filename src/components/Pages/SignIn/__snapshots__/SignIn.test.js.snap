// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignIn component should render correctly 1`] = `
<div>
  <div
    class="visually-hidden"
  >
    <img
      src="src/images/controllersImage.svg"
    />
  </div>
  <div
    class="user-page"
  >
    <header
      class="page-header user-page__head"
    >
      <div
        class="logo"
      >
        <a
          class="logo__link"
          href="/"
        >
          <span
            class="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            class="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            class="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        class="user-block"
      >
        <a
          class="user-block__link"
          href="/login"
        >
          Sign in
        </a>
      </div>
    </header>
    <div
      class="sign-in user-page__content"
    >
      <form
        action=""
        class="sign-in__form"
        method="POST"
        novalidate=""
      >
        <div
          class="sign-in__fields"
        >
          <div
            class="sign-in__field"
          >
            <input
              class="sign-in__input"
              data-testid="email"
              id="email"
              name="email"
              placeholder="Email address"
              type="email"
              value=""
            />
            <label
              class="sign-in__label visually-hidden"
              for="email"
            >
              Email address
            </label>
          </div>
          <div
            class="sign-in__field"
          >
            <input
              class="sign-in__input"
              data-testid="password"
              id="password"
              minlength="6"
              name="password"
              placeholder="Password"
              type="password"
              value=""
            />
            <label
              class="sign-in__label visually-hidden"
              for="password"
            >
              Password
            </label>
          </div>
        </div>
        <div
          class="sign-in__submit"
        >
          <button
            class="sign-in__btn"
            disabled=""
            type="submit"
          >
            Sign in
          </button>
        </div>
      </form>
    </div>
    <footer
      class="page-footer"
    >
      <div
        class="logo"
      >
        <a
          class="logo__link logo__link--light"
          href="/"
        >
          <span
            class="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            class="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            class="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        class="copyright"
      >
        <p>
          Â© 2019 What to watch Ltd.
        </p>
      </div>
    </footer>
  </div>
</div>
`;
